core/templates/pages/signup-page/signup-page.component.spec.ts

```diff
@@ -277,7 +277,7 @@ describe('Sign up page component', () => {
 
   it('should submit prerequisites form and save analytics', fakeAsync(() => {
     spyOn(urlService, 'getUrlParams').and.returnValue({
-      return_url: 'creator-dashboard'
+      return_url: 'contributor-dashboard'
```

Comment:
Can we have an explicit test case instead to check that the defaultDashboard is set correctly when the form is submitted?

Reply:
@kevintab95 Sure. Would something like this be okay?

```ts
it('should submit prerequisites form with user\'s preferred default dashboard', () => {
  spyOn(urlService, 'getUrlParams').and.returnValue({
    return_url: 'contributor-dashboard'
  });
  spyOn(signupPageBackendApiService, 'updateUsernameAsync');
  spyOn(siteAnalyticsService, 'registerNewSignupEvent');

  const sentRequestParams = {
    agreed_to_terms: true,
    can_receive_email_updates: false,
    default_dashboard: AppConstants.DASHBOARD_TYPE_CONTRIBUTOR,
    username: 'username'
  };
  
  componentInstance.hasUsername = false;
  componentInstance.showEmailPreferencesForm = true;

  componentInstance.submitPrerequisitesForm(
    sentRequestParams.agreed_to_terms,
    sentRequestParams.username, 'no');
  expect(signupPageBackendApiService.updateUsernameAsync).toHaveBeenCalledWith(
    sentRequestParams);
});
```